lib/logind.c

```diff
@@ -14,13 +14,13 @@
 
 #include <systemd/sd-login.h>
 
-int get_session_host (char **out)
+int get_session_host (char **out, bool is_forked)
 {
     char *host = NULL;
     char *session = NULL;
     int ret;
 
-    ret = sd_pid_get_session (getpid(), &session);
+    ret = sd_pid_get_session (is_forked ? getppid() : getpid(), &session);
     if (ret < 0) {
         return ret;
     }
```

Comment:
I think it would be simpler if you pass the `pid` to this function directly, instead of passing the boolean.

```c
int
get_session_host(char **out, pid_t pid)
{
    char *host = NULL;
    char *session = NULL;
    int ret;

    ret = sd_pid_get_session (pid, &session);
    if (ret < 0) {
        return ret;
}

Reply:
Right.
It is better to get PID just one time and save a few syscalls.